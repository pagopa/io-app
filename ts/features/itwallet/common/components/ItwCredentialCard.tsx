import {
  HStack,
  IOColors,
  makeFontStyleObject,
  Tag
} from "@pagopa/io-app-design-system";
import React from "react";
import { ImageSourcePropType, StyleSheet, Text, View } from "react-native";
import { AnimatedImage } from "../../../../components/AnimatedImage";
import I18n from "../../../../i18n";
import { getCredentialNameFromType } from "../utils/itwCredentialUtils";
import { CredentialType } from "../utils/itwMocksUtils";
import { getThemeColorByCredentialType } from "../utils/itwStyleUtils";
import { ItwDigitalVersionBadge } from "./ItwDigitalVersionBadge";

export type ItwCredentialStatus = "valid" | "pending" | "expiring" | "expired";

export type ItwCredentialCard = {
  credentialType: string;
  status?: ItwCredentialStatus;
  isPreview?: boolean;
};

export const ItwCredentialCard = ({
  status = "valid",
  credentialType,
  isPreview = false
}: ItwCredentialCard) => {
  const isValid = status === "valid";
  const theme = getThemeColorByCredentialType(credentialType);
  const labelColor = isValid ? theme.textColor : IOColors["grey-700"];

  const cardBackgroundSource =
    credentialCardBackgrounds[credentialType][isValid ? 0 : 1];
  const statusTagProps = tagPropsByStatus[status];

  return (
    <View style={isPreview && styles.previewContainer}>
      <View style={styles.cardContainer}>
        <View style={styles.card}>
          <AnimatedImage
            source={cardBackgroundSource}
            style={styles.cardBackground}
          />
        </View>
        <View style={styles.header}>
          <HStack space={16}>
            <Text style={[styles.label, { color: labelColor }]}>
              {getCredentialNameFromType(credentialType, "").toUpperCase()}
            </Text>
            {statusTagProps && <Tag {...statusTagProps} />}
          </HStack>
        </View>
        <ItwDigitalVersionBadge credentialType={credentialType} />
        <View
          style={[styles.border, { borderColor: borderColorByStatus[status] }]}
        />
      </View>
    </View>
  );
};

const credentialCardBackgrounds: {
  [type: string]: [ImageSourcePropType, ImageSourcePropType];
} = {
  [CredentialType.EUROPEAN_DISABILITY_CARD]: [
    require("../../../../../img/features/itWallet/cards/dc.png"),
    require("../../../../../img/features/itWallet/cards/dc_off.png")
  ],
  [CredentialType.EUROPEAN_HEALTH_INSURANCE_CARD]: [
    require("../../../../../img/features/itWallet/cards/ts.png"),
    require("../../../../../img/features/itWallet/cards/ts_off.png")
  ],
  [CredentialType.DRIVING_LICENSE]: [
    require("../../../../../img/features/itWallet/cards/mdl.png"),
    require("../../../../../img/features/itWallet/cards/mdl_off.png")
  ],
  [CredentialType.PID]: [
    require("../../../../../img/features/itWallet/cards/eid.png"),
    require("../../../../../img/features/itWallet/cards/eid_off.png")
  ]
};

const tagPropsByStatus: { [key in ItwCredentialStatus]?: Tag } = {
  expired: {
    variant: "error",
    text: I18n.t("features.itWallet.card.status.expired")
  },
  expiring: {
    variant: "warning",
    text: I18n.t("features.itWallet.card.status.expiring")
  },
  pending: {
    variant: "customIcon",
    text: I18n.t("features.itWallet.card.status.pending"),
    customIconProps: { iconColor: "info-700", iconName: "infoFilled" }
  }
};

const borderColorByStatus: { [key in ItwCredentialStatus]: string } = {
  valid: IOColors.white,
  expired: IOColors["error-600"],
  expiring: IOColors["warning-700"],
  pending: IOColors["info-700"]
};

const styles = StyleSheet.create({
  previewContainer: {
    aspectRatio: 9 / 2,
    overflow: "hidden"
  },
  cardContainer: {
    aspectRatio: 16 / 10,
    borderRadius: 8,
    overflow: "hidden"
  },
  card: {
    position: "absolute",
    left: 0,
    right: 0,
    top: 0,
    bottom: 0,
    backgroundColor: IOColors["grey-100"]
  },
  cardBackground: { height: "100%", width: "100%" },
  border: {
    position: "absolute",
    left: 0,
    right: 0,
    top: 0,
    bottom: 0,
    borderRadius: 8,
    borderWidth: 2
  },
  label: {
    flex: 1,
    ...makeFontStyleObject("Semibold", false, "TitilliumSansPro"),
    fontSize: 16
  },
  header: {
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
    paddingHorizontal: 16,
    paddingTop: 14
  }
});
