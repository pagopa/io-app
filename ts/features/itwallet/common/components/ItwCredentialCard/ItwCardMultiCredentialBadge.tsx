// The path data for the two shapes that make up the badge.
// These are extracted directly from the original SVG.
import { Canvas, Group, Path } from "@shopify/react-native-skia";

const PATH_BACK_CARD =
  "M0 4.8C0 3.80589 0.736564 3 1.64516 3H15.3548C16.2634 3 17 3.80589 17 4.8V13.2C17 14.1941 16.2634 15 15.3548 15H1.64516C0.736564 15 0 14.1941 0 13.2V4.8Z";
const PATH_FRONT_CARD =
  "M15.0762 3.33789L15.1592 3.33984C16.0197 3.38669 16.7041 4.15158 16.7041 5.08789V12.1709H5.60547C4.60285 12.1707 3.79012 11.3581 3.79004 10.3555V3.33789H15.0762ZM18.9746 0.170898C19.9773 0.171092 20.79 0.984592 20.79 1.9873V10.3555C20.79 11.3581 19.9772 12.1707 18.9746 12.1709H17.79V5.08789C17.79 3.47721 16.575 2.17114 15.0762 2.1709H3.79004V1.9873C3.79004 0.984592 4.6028 0.171093 5.60547 0.170898H18.9746Z";

const ORIGINAL_WIDTH = 23;
const ORIGINAL_HEIGHT = 15;

/**
 * A React Native component to display a multi-credential badge icon using Skia.
 * This version is drawn programmatically with Path components.
 */
export const ItwCardMultiCredentialBadge = () => (
  <Canvas style={{ width: ORIGINAL_WIDTH, height: ORIGINAL_HEIGHT }}>
    <Group>
      <Path path={PATH_BACK_CARD} color="#403F36" />
      <Path path={PATH_FRONT_CARD} color="#403C36" />
    </Group>
  </Canvas>
);
