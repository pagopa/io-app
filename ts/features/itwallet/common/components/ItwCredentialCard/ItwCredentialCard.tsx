import { HStack, IOText, Tag } from "@pagopa/io-app-design-system";
import Color from "color";
import { useMemo } from "react";
import { StyleSheet, View } from "react-native";
import { useIOSelector } from "../../../../../store/hooks";
import { fontPreferenceSelector } from "../../../../../store/reducers/persistedPreferences";
import {
  getCredentialNameFromType,
  tagPropsByStatus,
  useBorderColorByStatus,
  validCredentialStatuses
} from "../../utils/itwCredentialUtils";
import { getThemeColorByCredentialType } from "../../utils/itwStyleUtils";
import { ItwCredentialStatus } from "../../utils/itwTypesUtils";
import { CardBackground } from "./CardBackground";
import { DigitalVersionBadge } from "./DigitalVersionBadge";
import { CardColorScheme } from "./types";

export type ItwCredentialCard = {
  credentialType: string;
  status?: ItwCredentialStatus;
};

type StyleProps = {
  titleColor: string;
  titleOpacity: number;
  colorScheme: CardColorScheme;
};

export const ItwCredentialCard = ({
  status = "valid",
  credentialType
}: ItwCredentialCard) => {
  const typefacePreference = useIOSelector(fontPreferenceSelector);

  const borderColorMap = useBorderColorByStatus();
  const statusTagProps = tagPropsByStatus[status];

  const { titleColor, titleOpacity, colorScheme } = useMemo<StyleProps>(() => {
    const isValid = validCredentialStatuses.includes(status);
    const theme = getThemeColorByCredentialType(credentialType);

    if (status === "unknown") {
      return {
        titleColor: Color(theme.textColor).grayscale().hex(),
        titleOpacity: 0.5,
        colorScheme: "greyscale"
      };
    }

    if (isValid) {
      return {
        titleColor: theme.textColor,
        titleOpacity: 1,
        colorScheme: "default"
      };
    }

    return {
      titleColor: theme.textColor,
      titleOpacity: 0.5,
      colorScheme: "faded"
    };
  }, [credentialType, status]);

  return (
    <View style={styles.cardContainer}>
      <CardBackground
        credentialType={credentialType}
        colorScheme={colorScheme}
      />
      <View style={styles.header}>
        <HStack space={16}>
          <IOText
            size={16}
            lineHeight={20}
            font={
              typefacePreference === "comfortable"
                ? "Titillio"
                : "TitilliumSansPro"
            }
            weight="Semibold"
            maxFontSizeMultiplier={1.25}
            style={{
              letterSpacing: 0.25,
              color: titleColor,
              opacity: titleOpacity,
              flex: 1,
              flexShrink: 1
            }}
          >
            {getCredentialNameFromType(credentialType, "").toUpperCase()}
          </IOText>
          {statusTagProps && <Tag forceLightMode {...statusTagProps} />}
        </HStack>
      </View>
      <DigitalVersionBadge
        credentialType={credentialType}
        colorScheme={colorScheme}
      />
      <View style={[styles.border, { borderColor: borderColorMap[status] }]} />
    </View>
  );
};

const styles = StyleSheet.create({
  cardContainer: {
    aspectRatio: 16 / 10,
    borderRadius: 8,
    overflow: "hidden"
  },
  border: {
    position: "absolute",
    left: 0,
    right: 0,
    top: 0,
    bottom: 0,
    borderRadius: 8,
    borderCurve: "continuous",
    borderWidth: 2,
    zIndex: 11
  },
  header: {
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
    paddingHorizontal: 16,
    paddingTop: 14
  }
});
