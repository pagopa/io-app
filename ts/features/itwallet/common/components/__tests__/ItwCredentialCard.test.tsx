import { render } from "@testing-library/react-native";
import * as React from "react";
import { CredentialType } from "../../utils/itwMocksUtils";
import { ItwCredentialCard } from "../ItwCredentialCard";
import { ItwCredentialStatus } from "../../utils/itwTypesUtils";

describe("ItwCredentialCard", () => {
  it.each(["EuropeanHealthInsuranceCard", "EuropeanDisabilityCard", "MDL"])(
    "should match snapshot when credential type is %p",
    type => {
      const component = render(<ItwCredentialCard credentialType={type} />);
      expect(component).toMatchSnapshot();
    }
  );

  it.each([
    "valid",
    "expired",
    "expiring",
    "pending"
  ] as ReadonlyArray<ItwCredentialStatus>)(
    "should match snapshot when status is %p",
    status => {
      const component = render(
        <ItwCredentialCard credentialType={"MDL"} status={status} />
      );
      expect(component).toMatchSnapshot();
    }
  );

  it("should render the preview", () => {
    const component = render(
      <ItwCredentialCard
        credentialType={CredentialType.DRIVING_LICENSE}
        isPreview={true}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
