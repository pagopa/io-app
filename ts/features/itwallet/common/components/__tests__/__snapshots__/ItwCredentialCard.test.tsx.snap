// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItwCredentialCard should match snapshot when status is "expired" 1`] = `
<View
  style={false}
>
  <View
    style={
      {
        "aspectRatio": 1.6,
        "borderRadius": 8,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#E8EBF1",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        accessibilityIgnoresInvertColors={false}
        onLoad={[Function]}
        source={
          {
            "testUri": "../../../img/features/itWallet/cards/eid_off.png",
          }
        }
        style={
          [
            {
              "opacity": 0,
            },
            {
              "height": "100%",
              "width": "100%",
            },
          ]
        }
      />
    </View>
    <View
      style={
        {
          "paddingHorizontal": 16,
          "paddingTop": 14,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          allowFontScaling={false}
          color="grey-700"
          defaultColor="bluegrey"
          defaultWeight="Regular"
          dynamicTypeRamp="body"
          font="TitilliumSansPro"
          fontStyle={
            {
              "fontSize": 16,
              "lineHeight": 24,
            }
          }
          maxFontSizeMultiplier={1.25}
          numberOfLines={2}
          style={
            [
              {
                "flex": 1,
              },
              {
                "fontSize": 16,
                "lineHeight": 24,
              },
              {
                "color": "#555C70",
                "fontFamily": "Titillium Sans Pro",
                "fontStyle": "normal",
                "fontWeight": "600",
              },
            ]
          }
          weight="Semibold"
        >
          IDENTITÃ€ DIGITALE
        </Text>
        <View
          style={
            {
              "width": 16,
            }
          }
        />
        <View
          style={
            {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "backgroundColor": "#FFFFFF",
              "borderColor": "#E8EBF1",
              "borderCurve": "continuous",
              "borderRadius": 6,
              "borderWidth": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 8,
              "paddingVertical": 6,
            }
          }
        >
          <View
            style={
              {
                "flexShrink": 1,
              }
            }
          >
            <RNSVGSvgView
              accessibilityElementsHidden={true}
              accessibilityLabel=""
              accessible={false}
              align="xMidYMid"
              bbHeight={16}
              bbWidth={16}
              focusable={false}
              height={16}
              importantForAccessibility="no-hide-descendants"
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "color": "#D13333",
                  },
                  {
                    "flex": 0,
                    "height": 16,
                    "width": 16,
                  },
                ]
              }
              tintColor="#D13333"
              vbHeight={24}
              vbWidth={24}
              width={16}
            >
              <RNSVGGroup
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              >
                <RNSVGPath
                  clipRule={0}
                  d="M7.00702 1c-.73494 0-1.41405.39209-1.78152 1.02856L.2756 11.0274c-.36747.6365-.36747 1.4206 0 2.0571l4.9499 8.8783c.36747.6365 1.04658 1.0286 1.78152 1.0286h9.94288c.7349 0 1.414-.3921 1.7815-1.0286l4.993-8.8783c.3675-.6365.3675-1.4206 0-2.0571l-4.993-8.99884C18.3639 1.39209 17.6848 1 16.9499 1H7.00702Zm5.02608 13.5549c.5676 0 1.0283-.4601 1.0283-1.0277V5.52583c0-.56758-.4601-1.02769-1.0277-1.02769-.5676 0-1.0282.46011-1.0282 1.02769v8.00137c0 .5676.4601 1.0277 1.0276 1.0277Zm0 1.9431c.8436 0 1.5275.6839 1.5275 1.5275 0 .8436-.6839 1.5275-1.5275 1.5275-.8436 0-1.5274-.6839-1.5274-1.5275 0-.8436.6838-1.5275 1.5274-1.5275Z"
                  fill={
                    {
                      "type": 2,
                    }
                  }
                  fillRule={0}
                  propList={
                    [
                      "fill",
                      "fillRule",
                    ]
                  }
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <View
            style={
              {
                "width": 6,
              }
            }
          />
          <Text
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              [
                {
                  "alignSelf": "center",
                  "color": "#555C70",
                  "flexShrink": 1,
                  "fontSize": 12,
                  "lineHeight": 16,
                  "textTransform": "uppercase",
                },
                {
                  "fontFamily": "Titillium Sans Pro",
                  "fontStyle": "normal",
                  "fontWeight": "600",
                },
              ]
            }
          >
            Non valida
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "bottom": 16,
          "position": "absolute",
          "right": -10,
        }
      }
    >
      <View
        accessible={true}
        style={
          [
            {
              "alignItems": "center",
              "borderCurve": "continuous",
              "borderRadius": 24,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 8,
              "paddingVertical": 4,
            },
            {
              "backgroundColor": "#F4F5F8",
            },
          ]
        }
      >
        <Text
          accessibilityLabel="Versione digitale   "
          accessibilityRole="text"
          allowFontScaling={false}
          ellipsizeMode="tail"
          importantForAccessibility="yes"
          maxFontSizeMultiplier={1.25}
          numberOfLines={1}
          style={
            [
              {
                "alignSelf": "center",
                "flexShrink": 1,
                "fontSize": 12,
                "lineHeight": 16,
                "textTransform": "uppercase",
              },
              {
                "fontFamily": "Titillium Sans Pro",
                "fontStyle": "normal",
                "fontWeight": "600",
              },
              {
                "color": "#555C70",
              },
            ]
          }
        >
          Versione digitale   
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "borderColor": "transparent",
            "borderLeftWidth": 9,
            "borderRadius": 8,
            "borderWidth": 1,
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "borderColor": "#D13333",
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`ItwCredentialCard should match snapshot when status is "expiring" 1`] = `
<View
  style={false}
>
  <View
    style={
      {
        "aspectRatio": 1.6,
        "borderRadius": 8,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#E8EBF1",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        accessibilityIgnoresInvertColors={false}
        onLoad={[Function]}
        source={
          {
            "testUri": "../../../img/features/itWallet/cards/eid_off.png",
          }
        }
        style={
          [
            {
              "opacity": 0,
            },
            {
              "height": "100%",
              "width": "100%",
            },
          ]
        }
      />
    </View>
    <View
      style={
        {
          "paddingHorizontal": 16,
          "paddingTop": 14,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          allowFontScaling={false}
          color="grey-700"
          defaultColor="bluegrey"
          defaultWeight="Regular"
          dynamicTypeRamp="body"
          font="TitilliumSansPro"
          fontStyle={
            {
              "fontSize": 16,
              "lineHeight": 24,
            }
          }
          maxFontSizeMultiplier={1.25}
          numberOfLines={2}
          style={
            [
              {
                "flex": 1,
              },
              {
                "fontSize": 16,
                "lineHeight": 24,
              },
              {
                "color": "#555C70",
                "fontFamily": "Titillium Sans Pro",
                "fontStyle": "normal",
                "fontWeight": "600",
              },
            ]
          }
          weight="Semibold"
        >
          IDENTITÃ€ DIGITALE
        </Text>
        <View
          style={
            {
              "width": 16,
            }
          }
        />
        <View
          style={
            {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "backgroundColor": "#FFFFFF",
              "borderColor": "#E8EBF1",
              "borderCurve": "continuous",
              "borderRadius": 6,
              "borderWidth": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 8,
              "paddingVertical": 6,
            }
          }
        >
          <View
            style={
              {
                "flexShrink": 1,
              }
            }
          >
            <RNSVGSvgView
              accessibilityElementsHidden={true}
              accessibilityLabel=""
              accessible={false}
              align="xMidYMid"
              bbHeight={16}
              bbWidth={16}
              focusable={false}
              height={16}
              importantForAccessibility="no-hide-descendants"
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "color": "#A5822A",
                  },
                  {
                    "flex": 0,
                    "height": 16,
                    "width": 16,
                  },
                ]
              }
              tintColor="#A5822A"
              vbHeight={24}
              vbWidth={24}
              width={16}
            >
              <RNSVGGroup
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              >
                <RNSVGPath
                  clipRule={0}
                  d="m15.523 3.047 7.926 13.813c1.566 2.73-.392 6.14-3.523 6.14H4.074C.942 23-1.015 19.59.55 16.86L8.477 3.047c1.566-2.73 5.48-2.73 7.046 0Zm-3.495 13.458a1.528 1.528 0 1 1 0 3.056 1.528 1.528 0 0 1 0-3.056Zm0-1.944c.568 0 1.029-.46 1.029-1.028V5.528a1.028 1.028 0 0 0-2.057 0v8.005c0 .567.46 1.028 1.029 1.028Z"
                  fill={
                    {
                      "type": 2,
                    }
                  }
                  fillRule={0}
                  propList={
                    [
                      "fill",
                      "fillRule",
                    ]
                  }
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <View
            style={
              {
                "width": 6,
              }
            }
          />
          <Text
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              [
                {
                  "alignSelf": "center",
                  "color": "#555C70",
                  "flexShrink": 1,
                  "fontSize": 12,
                  "lineHeight": 16,
                  "textTransform": "uppercase",
                },
                {
                  "fontFamily": "Titillium Sans Pro",
                  "fontStyle": "normal",
                  "fontWeight": "600",
                },
              ]
            }
          >
            In scadenza
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "bottom": 16,
          "position": "absolute",
          "right": -10,
        }
      }
    >
      <View
        accessible={true}
        style={
          [
            {
              "alignItems": "center",
              "borderCurve": "continuous",
              "borderRadius": 24,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 8,
              "paddingVertical": 4,
            },
            {
              "backgroundColor": "#F4F5F8",
            },
          ]
        }
      >
        <Text
          accessibilityLabel="Versione digitale   "
          accessibilityRole="text"
          allowFontScaling={false}
          ellipsizeMode="tail"
          importantForAccessibility="yes"
          maxFontSizeMultiplier={1.25}
          numberOfLines={1}
          style={
            [
              {
                "alignSelf": "center",
                "flexShrink": 1,
                "fontSize": 12,
                "lineHeight": 16,
                "textTransform": "uppercase",
              },
              {
                "fontFamily": "Titillium Sans Pro",
                "fontStyle": "normal",
                "fontWeight": "600",
              },
              {
                "color": "#555C70",
              },
            ]
          }
        >
          Versione digitale   
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "borderColor": "transparent",
            "borderLeftWidth": 9,
            "borderRadius": 8,
            "borderWidth": 1,
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "borderColor": "#A5822A",
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`ItwCredentialCard should match snapshot when status is "pending" 1`] = `
<View
  style={false}
>
  <View
    style={
      {
        "aspectRatio": 1.6,
        "borderRadius": 8,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#E8EBF1",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        accessibilityIgnoresInvertColors={false}
        onLoad={[Function]}
        source={
          {
            "testUri": "../../../img/features/itWallet/cards/eid_off.png",
          }
        }
        style={
          [
            {
              "opacity": 0,
            },
            {
              "height": "100%",
              "width": "100%",
            },
          ]
        }
      />
    </View>
    <View
      style={
        {
          "paddingHorizontal": 16,
          "paddingTop": 14,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          allowFontScaling={false}
          color="grey-700"
          defaultColor="bluegrey"
          defaultWeight="Regular"
          dynamicTypeRamp="body"
          font="TitilliumSansPro"
          fontStyle={
            {
              "fontSize": 16,
              "lineHeight": 24,
            }
          }
          maxFontSizeMultiplier={1.25}
          numberOfLines={2}
          style={
            [
              {
                "flex": 1,
              },
              {
                "fontSize": 16,
                "lineHeight": 24,
              },
              {
                "color": "#555C70",
                "fontFamily": "Titillium Sans Pro",
                "fontStyle": "normal",
                "fontWeight": "600",
              },
            ]
          }
          weight="Semibold"
        >
          IDENTITÃ€ DIGITALE
        </Text>
        <View
          style={
            {
              "width": 16,
            }
          }
        />
        <View
          style={
            {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "backgroundColor": "#FFFFFF",
              "borderColor": "#E8EBF1",
              "borderCurve": "continuous",
              "borderRadius": 6,
              "borderWidth": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 8,
              "paddingVertical": 6,
            }
          }
        >
          <View
            style={
              {
                "flexShrink": 1,
              }
            }
          >
            <RNSVGSvgView
              accessibilityElementsHidden={true}
              accessibilityLabel=""
              accessible={false}
              align="xMidYMid"
              bbHeight={16}
              bbWidth={16}
              focusable={false}
              height={16}
              importantForAccessibility="no-hide-descendants"
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "color": "#418DAF",
                  },
                  {
                    "flex": 0,
                    "height": 16,
                    "width": 16,
                  },
                ]
              }
              tintColor="#418DAF"
              vbHeight={24}
              vbWidth={24}
              width={16}
            >
              <RNSVGGroup
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              >
                <RNSVGPath
                  clipRule={0}
                  d="M12 0C5.373 0 0 5.373 0 12s5.373 12 12 12 12-5.373 12-12S18.627 0 12 0Zm.028 7.556a1.528 1.528 0 1 0 0-3.056 1.528 1.528 0 0 0 0 3.056Zm0 1.944c.568 0 1.029.46 1.029 1.028v8.005a1.028 1.028 0 0 1-2.057 0v-8.005c0-.568.46-1.028 1.029-1.028Z"
                  fill={
                    {
                      "type": 2,
                    }
                  }
                  fillRule={0}
                  propList={
                    [
                      "fill",
                      "fillRule",
                    ]
                  }
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <View
            style={
              {
                "width": 6,
              }
            }
          />
          <Text
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              [
                {
                  "alignSelf": "center",
                  "color": "#555C70",
                  "flexShrink": 1,
                  "fontSize": 12,
                  "lineHeight": 16,
                  "textTransform": "uppercase",
                },
                {
                  "fontFamily": "Titillium Sans Pro",
                  "fontStyle": "normal",
                  "fontWeight": "600",
                },
              ]
            }
          >
            In lavorazione
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "bottom": 16,
          "position": "absolute",
          "right": -10,
        }
      }
    >
      <View
        accessible={true}
        style={
          [
            {
              "alignItems": "center",
              "borderCurve": "continuous",
              "borderRadius": 24,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 8,
              "paddingVertical": 4,
            },
            {
              "backgroundColor": "#F4F5F8",
            },
          ]
        }
      >
        <Text
          accessibilityLabel="Versione digitale   "
          accessibilityRole="text"
          allowFontScaling={false}
          ellipsizeMode="tail"
          importantForAccessibility="yes"
          maxFontSizeMultiplier={1.25}
          numberOfLines={1}
          style={
            [
              {
                "alignSelf": "center",
                "flexShrink": 1,
                "fontSize": 12,
                "lineHeight": 16,
                "textTransform": "uppercase",
              },
              {
                "fontFamily": "Titillium Sans Pro",
                "fontStyle": "normal",
                "fontWeight": "600",
              },
              {
                "color": "#555C70",
              },
            ]
          }
        >
          Versione digitale   
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "borderColor": "transparent",
            "borderLeftWidth": 9,
            "borderRadius": 8,
            "borderWidth": 1,
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "borderColor": "#418DAF",
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`ItwCredentialCard should match snapshot when status is "valid" 1`] = `
<View
  style={false}
>
  <View
    style={
      {
        "aspectRatio": 1.6,
        "borderRadius": 8,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#E8EBF1",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        accessibilityIgnoresInvertColors={false}
        onLoad={[Function]}
        source={
          {
            "testUri": "../../../img/features/itWallet/cards/eid.png",
          }
        }
        style={
          [
            {
              "opacity": 0,
            },
            {
              "height": "100%",
              "width": "100%",
            },
          ]
        }
      />
    </View>
    <View
      style={
        {
          "paddingHorizontal": 16,
          "paddingTop": 14,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          allowFontScaling={false}
          color="bluegreyDark"
          defaultColor="bluegrey"
          defaultWeight="Regular"
          dynamicTypeRamp="body"
          font="TitilliumSansPro"
          fontStyle={
            {
              "fontSize": 16,
              "lineHeight": 24,
            }
          }
          maxFontSizeMultiplier={1.25}
          numberOfLines={2}
          style={
            [
              {
                "flex": 1,
              },
              {
                "fontSize": 16,
                "lineHeight": 24,
              },
              {
                "color": "#17324D",
                "fontFamily": "Titillium Sans Pro",
                "fontStyle": "normal",
                "fontWeight": "600",
              },
            ]
          }
          weight="Semibold"
        >
          IDENTITÃ€ DIGITALE
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "borderColor": "transparent",
            "borderLeftWidth": 9,
            "borderRadius": 8,
            "borderWidth": 1,
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "borderColor": "transparent",
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`ItwCredentialCard should render the preview 1`] = `
<View
  style={
    {
      "aspectRatio": 4.5,
      "overflow": "hidden",
    }
  }
>
  <View
    style={
      {
        "aspectRatio": 1.6,
        "borderRadius": 8,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#E8EBF1",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        accessibilityIgnoresInvertColors={false}
        onLoad={[Function]}
        source={
          {
            "testUri": "../../../img/features/itWallet/cards/eid.png",
          }
        }
        style={
          [
            {
              "opacity": 0,
            },
            {
              "height": "100%",
              "width": "100%",
            },
          ]
        }
      />
    </View>
    <View
      style={
        {
          "paddingHorizontal": 16,
          "paddingTop": 14,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          allowFontScaling={false}
          color="bluegreyDark"
          defaultColor="bluegrey"
          defaultWeight="Regular"
          dynamicTypeRamp="body"
          font="TitilliumSansPro"
          fontStyle={
            {
              "fontSize": 16,
              "lineHeight": 24,
            }
          }
          maxFontSizeMultiplier={1.25}
          numberOfLines={2}
          style={
            [
              {
                "flex": 1,
              },
              {
                "fontSize": 16,
                "lineHeight": 24,
              },
              {
                "color": "#17324D",
                "fontFamily": "Titillium Sans Pro",
                "fontStyle": "normal",
                "fontWeight": "600",
              },
            ]
          }
          weight="Semibold"
        >
          IDENTITÃ€ DIGITALE
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "borderColor": "transparent",
            "borderLeftWidth": 9,
            "borderRadius": 8,
            "borderWidth": 1,
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "borderColor": "transparent",
          },
        ]
      }
    />
  </View>
</View>
`;
